gno.land/pkg/sdk/vm/keeper.go

```diff
@@ -1204,10 +1204,10 @@ func (vm *VMKeeper) lockStorageDeposit(ctx sdk.Context, caller crypto.Address, r
 	return nil
 }
 
-func (vm *VMKeeper) refundStorageDeposit(ctx sdk.Context, caller crypto.Address, rlm *gno.Realm, depositUnlocked int64, released int64) error {
+func (vm *VMKeeper) refundStorageDeposit(ctx sdk.Context, storageFeeCollector crypto.Address, rlm *gno.Realm, depositUnlocked int64, released int64) error {
 	storageDepositAddr := gno.DeriveStorageDepositCryptoAddr(rlm.Path)
 	d := std.Coins{std.Coin{Denom: ugnot.Denom, Amount: depositUnlocked}}
-	err := vm.bank.SendCoinsUnrestricted(ctx, storageDepositAddr, caller, d)
+	err := vm.bank.SendCoinsUnrestricted(ctx, storageDepositAddr, storageFeeCollector, d)
```

Comment:
This changes to not send the refund to the caller. But `processStorageDeposit` still [emits a `StorageUnlockEvent`](https://github.com/gnolang/gno/blob/4dbfb49c98cd93d9dc61d2d8916605d8f26f0cdd/gno.land/pkg/sdk/vm/keeper.go#L1175-L1181), which gnokey uses to update the caller's transaction cost (assuming that the caller receives the refund). I think this PR needs to update the [calculation of the transaction cost](https://github.com/gnolang/gno/blob/71e9ac3593a9eb51446e92ca908e3ddeda712513/gno.land/pkg/keyscli/root.go#L79) to ignore a refund. I can help if you want.